ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces/utils.py

```diff
@@ -0,0 +1,352 @@
+        ip = get(p2p_link, "ip", default=[None, None])
+        # Set bgp_as or fallback to list with None values
+        bgp_as = get(p2p_link, "as", default=[None, None])
+
+        data = {
+            "peer": peer,
+            "peer_type": peer_type,
+            "ip": ip[index],
+            "peer_ip": ip[peer_index],
+            "bgp_as": str(bgp_as[index]),
+            "peer_bgp_as": str(bgp_as[peer_index]),
+        }
+
+        node_child_interfaces = get(p2p_link, "port_channel.nodes_child_interfaces")
+        # Convert to new data models
+        node_child_interfaces = convert_dicts(node_child_interfaces, primary_key="node", secondary_key="interfaces")
+        if member_interfaces := get_item(node_child_interfaces, "node", self._hostname, default={}).get("interfaces"):
+            # Port-channel
+            peer_member_interfaces = get_item(
+                node_child_interfaces,
+                "node",
+                peer,
+                required=True,
+                var_name=f"{peer} under core_interfaces.p2p_links.[].port_channel.nodes_child_interfaces",
+            )["interfaces"]
+            id = int("".join(re.findall(r"\d", member_interfaces[0])))
+            peer_id = int("".join(re.findall(r"\d", peer_member_interfaces[0])))
+            data.update(
+                {
+                    "interface": f"Port-Channel{id}",
+                    "peer_interface": f"Port-Channel{peer_id}",
+                    "port_channel_id": id,
+                    "port_channel_members": [
+                        {
+                            "interface": interface,
+                            "peer_interface": peer_member_interfaces[index],
+                        }
+                        for index, interface in enumerate(member_interfaces)
+                    ],
+                }
+            )
+            return data
+
+        if "interfaces" in p2p_link:
+            # Ethernet
+            data.update(
+                {
+                    "interface": p2p_link["interfaces"][index],
+                    "peer_interface": p2p_link["interfaces"][peer_index],
+                    "port_channel_id": None,
+                    "port_channel_members": [],
+                }
+            )
+            return data
+
+        raise AristaAvdMissingVariableError("core_interfaces.p2p_links must have either 'interfaces' or 'port_channel' with correct members set.")
+
+    def _get_common_interface_cfg(self, p2p_link: dict) -> dict:
+        """
+        Return partial structured_config for one p2p_link.
+        Covers common config that is applicable to both port-channels and ethernet interfaces.
+        This config will only be used on the main interface - so not port-channel members.
+        """
+
+        index = p2p_link["nodes"].index(self._hostname)
+        peer = p2p_link["data"]["peer"]
+        peer_interface = p2p_link["data"]["peer_interface"]
+        ethernet_interface = {
```

Comment:
maybe let's not call this variable `ethernet_interface` as it could be also a port_channel interface

Reply:
good point. I will fix this

Reply:
Fixed in https://github.com/aristanetworks/ansible-avd/pull/2284/commits/674860ba30d38cf8e7a8848f96cd26b5577413fe